### PR TITLE
upgrade Android-Components to 0.25.1

### PR DESCRIPTION
**Pull request by linuxwolf**
Updates our Android Components dependencies to the latest published (0.25.1 as of 10-01).  This is needed to pick up on the most up-to-date version of `LoginsStorage`.

### CODE REVIEW

File: app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt [hunk 1]
```diff
@@ -43,6 +46,30 @@ open class MockLoginsStorage : LoginsStorage {
...
+
+            all.add(login)
+            SyncResult.fromValue(login.id)
+        }
+    }
+
+    override fun update(login: ServerPassword): SyncResult<Unit> {
+        return list().then {
+            val position = it.indexOfFirst { it.id == login.id }
+            if (position == -1) {
```
Comment (jhugman) [inline app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt]: Nit: prefer 

a) putting the positive branch first and the exceptional branch second.
b) using the numbers `0`, `1` and `n`.

i.e.

```kotlin
if (position >= 0) {

} else {
    throw …
}
```

File: app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt [hunk 2]
```diff
@@ -43,6 +46,30 @@ open class MockLoginsStorage : LoginsStorage {
...
     }
+    override fun add(login: ServerPassword): SyncResult<String> {
+        return list().then {
+            if (login.id == "") {
+                SyncResult.fromException<String>(InvalidRecordException("record id missing"))
+            } else if (null != it.find { it.id == login.id }) {
+                SyncResult.fromException<String>(IdCollisionException("record already exists"))
+            }
+
+            all.add(login)
```
Comment (jhugman) [inline app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt]: What is `all`?

File: app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt [hunk 3]
```diff
@@ -43,6 +46,30 @@ open class MockLoginsStorage : LoginsStorage {
             SyncResult.fromValue(true)
         }
     }
+    override fun add(login: ServerPassword): SyncResult<String> {
+        return list().then {
+            if (login.id == "") {
+                SyncResult.fromException<String>(InvalidRecordException("record id missing"))
+            } else if (null != it.find { it.id == login.id }) {
+                SyncResult.fromException<String>(IdCollisionException("record already exists"))
+            }
```
Comment (jhugman) [inline app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt]: This logic needs a little bit of re-work. 

Can we use `get(login.id).then {}`

Can we put the empty string check as a guard condition at the top of the method?

File: app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt [hunk 4]
```diff
@@ -43,6 +46,30 @@ open class MockLoginsStorage : LoginsStorage {
...
+
+            all.add(login)
+            SyncResult.fromValue(login.id)
+        }
+    }
+
+    override fun update(login: ServerPassword): SyncResult<Unit> {
+        return list().then {
```
Comment (jhugman) [inline app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt]: Again, use `get()`

File: app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt (hunk 2 above)
Comment (linuxwolf) [reply app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt]: all is the MutableList of credential instances.

Comment (jhugman) [reply app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt]: 👍

**Pull request merged**